### ISSUE TITLE
`unicorn/no-invalid-remove-event-listener` does not report when optional chaining

### ISSUE THREAD

**Issue by mcenkar** [sindresorhus/eslint-plugin-unicorn#2253]
Similar to #2252 but on `unicorn/no-invalid-remove-event-listener`

Rule works correctly without optional chaining:

<img width="1158" alt="image" src="https://github.com/sindresorhus/eslint-plugin-unicorn/assets/11230749/2067ab12-90aa-496a-95a3-d5b3d4e0f7e2">

However with optional chaining stops reporting error:

<img width="678" alt="image" src="https://github.com/sindresorhus/eslint-plugin-unicorn/assets/11230749/db80e3bc-2ffd-4ff2-bbcf-ec1c093ee4f2">

```js
this.input?.removeEventListener(event, this.onInputEvent.bind(this), true)
```

Can be easily tested by adding `?` after `el`: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/f0ff04d/test/no-invalid-remove-event-listener.mjs#L49
